models/model_choice.py

```diff
@@ -68,6 +71,55 @@ def load_checkpoint(filename):
         raise FileNotFoundError(f"=> No model found at '{filename}'")
 
 
+def verify_weights(num_classes, weights):
+    """Verifies that the number of weights equals the number of classes if any are given
+    Args:
+        num_classes: number of classes defined in the configuration file
+        weights: weights defined in the configuration file
+    """
+    if num_classes == 1 and len(weights) == 2:
+        warnings.warn("got two class weights for single class defined in configuration file; will assume index 0 = background")
+    elif num_classes != len(weights):
+        raise ValueError('The number of class weights in the configuration file is different than the number of classes')
+
+
+def set_hyperparameters(params, num_classes, model, checkpoint, dontcare_val):
```

Comment:
We have a `set_hyperparameters()` function [in train_segmentation.py](https://github.com/NRCan/geo-deep-learning/blob/6852729d0ec03c0c2a6465b7a68039181c8bcbda/train_segmentation.py#L147). Though it shouldn't be a problem for autocompletion, for example, in our IDEs, should we consider a different name? ex.: `set_model_hyperparam()`